src/constructLayoutEngine.js

```diff
@@ -63,56 +72,97 @@ export function constructLayoutEngine({
   function arrangeDomElements() {
     const path = location[resolvedRoutes.mode === "hash" ? "hash" : "pathname"];
 
-    // TODO memoize this for perf
-    const matchedRoutes = matchRoute(resolvedRoutes, path);
+    if (!path.startsWith(baseWithoutSlash)) {
+      // Base URL doesn't match, no need to recurse routes
+      return;
+    }
 
-    const initialContainer =
+    const parentContainer =
       typeof resolvedRoutes.containerEl === "string"
         ? document.querySelector(resolvedRoutes.containerEl)
         : resolvedRoutes.containerEl;
 
-    recurseRoutes(matchedRoutes.routes, initialContainer);
+    recurseRoutes({
```

Comment:
This function now needs more arguments because it now needs to check whether routes match the current path.